cdap-ui/app/cdap/components/PipelineSummary/LogsMetricsGraph/index.js

```diff
@@ -0,0 +1,291 @@
+    }
+    let popOverData, logUrl;
+    if (this.state.currentHoveredElement) {
+      popOverData = this.props.runs.find(run => this.state.currentHoveredElement.runid === run.runid);
+      let {namespaceId, appId, programType, programId} = this.props.runContext;
+      logUrl = `/logviewer/view?namespace=${namespaceId}&appId=${appId}&programType=${convertProgramToApi(programType)}&programId=${programId}&runId=${popOverData.runid}`;
+    }
+    if (this.props.isLoading) {
+      return (
+        <div className="empty-runs-container">
+          <IconSVG
+            name="icon-spinner"
+            className="fa-spin"
+          />
+        </div>
+      );
+    }
+    if (!this.props.runs.length) {
+      return (
+        <div className="empty-runs-container">
+          <h1> {T.translate(`${GRAPHPREFIX}.emptyMessage`)} </h1>
+        </div>
+      );
+    }
+    return (
+      <div className="graph-plot-container">
+        <FPlot
+          className="logs-metrics-fp-plot"
+          xType="linear"
+          xDomain={xDomain}
+          stackBy="y"
+          height={height}>
+          <DiscreteColorLegend
+            style={{position: 'absolute', left: '40px', top: '0px'}}
+            orientation="horizontal" items={[
+              {
+                title: T.translate(`${PREFIX}.legend1`),
+                color: WARNINGBARCOLOR
+              },
+              {
+                title: T.translate(`${PREFIX}.legend2`),
+                color: ERRORBARCOLOR
+              }
+            ]}
+          />
+          <HorizontalGridLines />
+          <XAxis
+            tickTotal={10}
+            tickFormat={(v => {
+              if (this.props.xDomainType === 'time') {
+                return moment(v * 1000).format('ddd M/D/YY');
+              }
+              return v;
+            })}
+          />
+          <YAxis />
+          <BarSeries
+            cluster="runs"
+            color="#FDA639"
+            onValueClick={(d) => {
+              if (isEqual(this.state.currentHoveredElement || {}, d)) {
+                this.setState({
+                  currentHoveredElement: null
+                });
+              } else {
+                this.setState({
+                  currentHoveredElement: d
+                });
+              }
+            }}
+            data={warnings}/>
+          <BarSeries
+            cluster="runs"
+            color="#A40403"
+            onValueClick={(d) => {
+              if (isEqual(this.state.currentHoveredElement || {}, d)) {
+                this.setState({
+                  currentHoveredElement: null
+                });
+              } else {
+                this.setState({
+                  currentHoveredElement: d
+                });
+              }
+            }}
+            data={errors}/>
+          {
+            this.state.currentHoveredElement && popOverData ?
+              <Hint value={this.state.currentHoveredElement}>
```

Comment:
Seems like a lot of inconsistent spacing between opening and closing tags in this block of code. Also, let's wrap this block inside a `( )`.